README.md

```diff
@@ -1,51 +1,99 @@
+mkdir -p ~/ros2_ws/src
+cd ~/ros2_ws/src
 git clone https://github.com/mangdangroboticsclub/mini_pupper_ros.git -b ros2
-git clone --recursive https://github.com/chvmp/champ -b ros2
-cd ..
+vcs import < minipupper_ros/.minipupper.repos --recursive
+```
+
+Build and install all ROS packages.
+
+```sh
+cd ~/ros2_ws
 rosdep install --from-paths src --ignore-src -r -y
+sudo apt-get install ros-humble-teleop-twist-keyboard
 colcon build --symlink-install
+```
+
+### 1.2 Mini Pupper Setup
+
+Mini Pupper Setup corresponds to the Raspberry Pi on your Mini Pupper.
+
+Ubuntu 22.04 + ROS 2 Humble is required.  
+Please follow the [installation document for ROS Humble](https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html) or use the [unofficial ROS 2 installation script](https://github.com/Tiryoh/ros2_setup_scripts_ubuntu).
+
+After ROS 2 installation, download the Mini Pupper ROS package in the workspace.
+
+```sh
+mkdir -p ~/ros2_ws/src
+cd ~/ros2_ws/src
+git clone https://github.com/mangdangroboticsclub/mini_pupper_ros.git -b ros2
+vcs import < minipupper_ros/.minipupper.repos --recursive
+# compiling gazebo and cartographer on Raspberry Pi is not recommended
+touch champ/champ/champ_description/AMENT_IGNORE
+touch champ/champ/champ_gazebo/AMENT_IGNORE
+touch champ/champ/champ_navigation/AMENT_IGNORE
+touch mini_pupper_ros/mini_pupper_gazebo/AMENT_IGNORE
+touch mini_pupper_ros/mini_pupper_navigation/AMENT_IGNORE
+```
+
+Build and install all ROS packages.
+
+If the Raspberry Pi has less than 4GB memory, try `MAKEFLAGS=-j1 colcon build --executor sequential --symlink-install` instead of `colcon build --symlink-install`
+
+```sh
+# install dependencies without unused heavy packages
+cd ~/ros2_ws
+rosdep install --from-paths src --ignore-src -r -y --skip-keys=joint_state_publisher_gui --skip-keys=rviz2
 sudo apt-get install ros-humble-teleop-twist-keyboard
+colcon build --symlink-install
 ```
 
-## 2. Test in RViz
+## 2. Quick Start
+
+### 2.1 Test in RViz
+
 ```sh
 # Terminal 1
 . ~/ros2_ws/install/setup.bash # setup.zsh if you use zsh instead of bash
-ros2 launch mini_pupper_bringup bringup.launch.py
+ros2 launch mini_pupper_bringup bringup.launch.py rviz:=true
 
 # Terminal 2
-. ~/ros2_ws/install/setup.bash
-rviz2 -d src/champ/champ_description/rviz/urdf_viewer.rviz
-
-# Terminal 3
 ros2 run teleop_twist_keyboard teleop_twist_keyboard
 # Then control robot dog with your keyboard
 ```
 
-## 3. Test in Gazebo
+### 2.2 Test in Gazebo
+
 ```sh
 # Terminal 1
 . ~/ros2_ws/install/setup.bash # setup.zsh if you use zsh instead of bash
-ros2 launch mini_pupper_gazebo gazebo.launch.py
+ros2 launch mini_pupper_gazebo gazebo.launch.py rviz:=true
 
 # Terminal 2
-. ~/ros2_ws/install/setup.bash
-rviz2 -d src/champ/champ_description/rviz/urdf_viewer.rviz
-
-# Terminal 3
 ros2 run teleop_twist_keyboard teleop_twist_keyboard
 # Then control robot dog with your keyboard
 ```
 
-## 4. Cartographer Test in Gazebo
+### 2.3 Cartographer Test in Gazebo
```

Comment:
I see a PR from @CullenSUN to CHAMP repo with `slam-toolbox` support. So do we still plan to use Cartographer?

Reply:
Yes, we use cartographer for now. We may change it later, but not in this pull request.

Reply:
ok, sounds good